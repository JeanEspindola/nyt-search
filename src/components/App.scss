.App {
  .App__Body{
    max-width: 900px;
    margin: 0 auto;
  }
}