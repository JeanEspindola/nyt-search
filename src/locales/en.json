{
  "go_to_results": "Go to Results Page",
  "header_title": "\"The New York Times\" article search application",
  "next_page": "Next Page",
  "no_details": "No details to show.",
  "no_results": "There are no results to display.",
  "previous_page": "Previous Page",
  "read_article": "Read the Full Article",
  "results": "Results:",
  "search_terms": "Search your terms here.",
  "type_query": "Type search query term in here:"
}