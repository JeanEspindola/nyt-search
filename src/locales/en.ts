// eslint-disable-next-line import/no-anonymous-default-export
export default {
  goToResults: "Go to Results Page",
  headerTitle: "\"The New York Times\" article search application",
  nextPage: "Next Page",
  noDetails: "No details to show.",
  noResults: "There are no results to display.",
  previousPage: "Previous Page",
  readArticle: "Read the Full Article",
  results: "Results:",
  searchTerms: "Search your terms here.",
  typeQuery: "Type search query term in here:"
}
