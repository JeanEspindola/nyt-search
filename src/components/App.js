/*
* App
*
* Initialize the UI rendering the Header and the MainrRoutes where the rest of
* application will be rendered based on the route.
*
* */
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Header from './Header/Header';
import MainRoutes from './Main-Routes/MainRoutes';
import en from '../locales/en.json';
import pt from '../locales/pt-br.json';

const locales = { en, pt };

class App extends Component {
  static childContextTypes = {
    locale: PropTypes.object,
  };

  getChildContext() {
    return {
      locale: locales.en,
    };
  }

  render() {
    return (
      <div className="App">
        <Header />
        <MainRoutes />
      </div>
    );
  }
}

export default App;
