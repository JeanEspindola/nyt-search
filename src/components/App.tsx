/*
* App
*
* Initialize the UI rendering the Header and the MainrRoutes where the rest of
* application will be rendered based on the route.
*
* */
import 'bootstrap/dist/css/bootstrap.min.css';
import { IntlProvider } from 'react-intl';
import { translationMessages } from '../locales';
import Header from './Header/Header';
import MainRoutes from './Main-Routes/MainRoutes';

const App = () => {
  return (
    <IntlProvider
      key="en"
      locale="en"
      defaultLocale="en"
      messages={translationMessages['en']}
    >
      <Header />
      <MainRoutes />
    </IntlProvider>
  );
}

export default App;
