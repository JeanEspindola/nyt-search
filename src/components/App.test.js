import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from './App';
import en from '../locales/en.json';

describe('App', () => {
  let Component;

  beforeEach(() => {
    Component = shallow(<App />);
  });

  it('renders App without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });

  it('should call getChildContext and return current locale', () => {
    expect(Component.instance().getChildContext().locale).toEqual(en);
  });
});
