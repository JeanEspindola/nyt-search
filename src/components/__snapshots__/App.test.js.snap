// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App without crashing 1`] = `
<React.Fragment>
  <connectContext />
  <MainRoutes />
</React.Fragment>
`;
